core/src/main/java/org/infinispan/remoting/transport/RetryOnFailureXSiteCommand.java

```diff
@@ -18,16 +18,16 @@
  * @author Pedro Ruivo
  * @since 7.0
  */
-public class RetryOnFailureXSiteCommand {
+public class RetryOnFailureXSiteCommand<O> {
 
    public static final RetryPolicy NO_RETRY = new MaxRetriesPolicy(0);
    private static final Log log = LogFactory.getLog(RetryOnFailureXSiteCommand.class);
    private static final boolean trace = log.isTraceEnabled();
    private final XSiteBackup xSiteBackup;
-   private final XSiteReplicateCommand command;
+   private final XSiteReplicateCommand<O> command;
    private final RetryPolicy retryPolicy;
 
-   private RetryOnFailureXSiteCommand(XSiteBackup backup, XSiteReplicateCommand command, RetryPolicy retryPolicy) {
+   private RetryOnFailureXSiteCommand(XSiteBackup backup, XSiteReplicateCommand<O> command, RetryPolicy retryPolicy) {
```

Comment:
can you add the generics to `org.infinispan.remoting.transport.RetryOnFailureXSiteCommand#newInstance()`?
just to remove the warnings from IDEA

Reply:
Sure, just missed that one.